ash/src/extensions/ext/drm_format_modifier.rs

```diff
@@ -0,0 +1,83 @@
+use crate::prelude::*;
+use crate::vk;
+use crate::{Device, Instance};
+use std::ffi::CStr;
+use std::mem;
+
+#[derive(Clone)]
+pub struct DrmFormatModifier {
+    handle: vk::Device,
+    fp: vk::ExtImageDrmFormatModifierFn,
+}
+
+impl DrmFormatModifier {
+    pub fn new(instance: &Instance, device: &Device) -> Self {
+        let handle = device.handle();
+        let fp = vk::ExtImageDrmFormatModifierFn::load(|name| unsafe {
+            mem::transmute(instance.get_device_proc_addr(handle, name.as_ptr()))
+        });
+        Self { handle, fp }
+    }
+
+    /// <https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/vkGetImageDrmFormatModifierPropertiesEXT.html>
+    pub unsafe fn get_image_drm_format_modifier_properties(
+        &self,
+        image: &vk::Image,
+        properties: &mut vk::ImageDrmFormatModifierPropertiesEXT,
+    ) -> VkResult<()> {
+        (self.fp.get_image_drm_format_modifier_properties_ext)(self.handle, *image, properties)
+            .result()
+    }
+
+    pub unsafe fn get_drm_format_properties_list(
+        instance: &Instance,
+        pdevice: vk::PhysicalDevice,
+        format: vk::Format,
+        format_properties: vk::FormatProperties,
+    ) -> Vec<vk::DrmFormatModifierPropertiesEXT> {
+        let mut list = vk::DrmFormatModifierPropertiesListEXT::default();
+
+        // Get count of DRM format modifiers for the format.
+        {
+            let mut format_properties_2 = vk::FormatProperties2::builder()
+                .format_properties(format_properties)
+                .push_next(&mut list);
```

Comment:
Just like other typical list getter functions, how will you let the caller append more `pNext` structs to `vk::DrmFormatModifierPropertiesListEXT`?

Reply:
I can't pass `&mut FormatProperties2` into the function since the `push_next` function does not exist on those structs.

I could emulate it via copying the content of FormatProperties2Builder::push_next, but that does not sound very maintainable.

Taking a builder feels odd but would be least prone to mistakes.

Reply:
It appears in experimentation that even if I take ownership of a `FormatProperties2Builder`, I end up with some of the borrowing issues I have described in the comment about using the helper functions.

Reply:
The recent changes to remove builders have made this possible now.